checks/raw/branch_protection.go

```diff
@@ -15,37 +15,56 @@
 package raw
 
 import (
-	"errors"
 	"fmt"
 	"regexp"
 
 	"github.com/ossf/scorecard/v4/checker"
 	"github.com/ossf/scorecard/v4/clients"
-	sce "github.com/ossf/scorecard/v4/errors"
 )
 
 const master = "master"
 
-type branchMap map[string]*clients.BranchRef
+var commit = regexp.MustCompile("^[a-f0-9]{40}$")
+
+type branchSet struct {
+	exists map[string]bool
```

Comment:
I'm curious if i'd work with just:
```
type branchSet map[string]clients.BranchRef
```

Reply:
That's what we had before and we finally iterated through all the map values to create a slice. Just wanted to avoid that with this.

Reply:
gotcha